### PR TITLE
 Anomaly core modules for modsuits

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Adds the shielded component.
Shielded hardsuits are gone, now a module for modsuits.
Gamma shielded hardsuit replaced with a hos modsuit with shielded module.
Nukies now need to pay a bit more for their shields, as they can now apply it to an elite suit. Now 200 tc from 150.

Adds an anomaly core module for each anomaly core type. 
Pyro gets the firewall module. This module makes a drop wall (20 second cooldown), where any projectile that passes through it gets a stack of immolate. 
Gamma sec ERT replace the mirage grenade with this. Gives them some needed protection and lets them heat things up on the battlefield.
Tesla gets the arc-shield module. A slowly regenerating shield, that regenerates 20 hp every 10 seconds. Will shock in melee, and occassionally arc when hit. 80 max HP. Disablers lower it.
Cryo gets the Cryonic grenade module. Shoots a cryogenic grenade that slips the floor, does a bit of stamina damage, traps people in ice and puts some frost oil and ice to keep em cool. Good for stopping people chasing you, but they can still melee / attack, and can break out faster by resisting.
Vortex gets the vortex shotgun module (name pending)
A wrist mounted energy shotgun that rips apart space in front of it. High spread, low range, but will fuck up whatever you hit. Strong, but competing to the vortex arm and stealth armor.

## Why It's Good For The Game

Module for each core is good! Shielded component makes things better

## Todo:

- [x] Vortex module
- [x] Cryo module, cryo grenade a la doom eternal
- [x] several message admins as breakpoints no worky
- [x] todo: Vortex shotgun sprite

## Testing

testing all the modules

## Changelog
:cl:
todo
/:cl:

### CODE REVIEW

**Comment by Admiral-Gigaton** [comment]
shotgun isnt as powerful as pulse gun, yes? also, screenshots/videos? would be good for seeing what a "vortex shotgun" or "cryo grenade" does. 

also, with how much work a cryo core needs, why not make it, say, a freeze ray? it slows people, but has a decent cooldown/power usage, and if you manage to hit em enough, it freezes them in place? (ice block effect, freezes people in place, but i THINK they can still use items, dont know fo sho) 

you can also add multiple modules for one core, on /tg/ (i dont know if we have it here, too rare to know) theres an anti-grav module and kinesis module (deadspace) which are made from grav cores, so if you had anything else that wasnt too god tier to add, you could.

**Comment by S34NW** [comment]
PR is will WIP, skipping map queue until ready

File: code/game/machinery/deployable.dm [hunk 1]
```diff
@@ -326,6 +327,8 @@
 	var/protected = FALSE
 	///The core shield that protects the generator
 	var/obj/structure/barricade/dropwall/core_shield = null
+	/// The type of dropwall
+	var/our_type = /obj/structure/barricade/dropwall
```
Comment (Sirryan2002) [inline code/game/machinery/deployable.dm:331]: name this `barricade_type` or somethin

File: code/modules/mob/living/simple_animal/hostile/mining/basilisk.dm [hunk 2]
```diff
@@ -119,14 +119,7 @@
...
 	temperature = 500 //Heats you up!
-
-/obj/item/projectile/temp/basilisk/magmawing/on_hit(atom/target, blocked = FALSE)
-	. = ..()
-	if(.)
-		var/mob/living/L = target
-		if(istype(L))
-			L.adjust_fire_stacks(0.1)
-			L.IgniteMob()
+	immolate = 1
```
Comment (Sirryan2002) [inline code/modules/mob/living/simple_animal/hostile/mining/basilisk.dm:122]: is this meant to be a boolean?

File: code/modules/mod/modules/modules_security.dm [hunk 3]
```diff
@@ -166,3 +166,126 @@
...
+	thrower = null
+	return ..()
+
+/obj/item/grenade/cryogrenade_mod/attack_self(mob/user)
+	. = ..()
+	thrower = user
+
+/obj/item/grenade/cryogrenade_mod/prime()
+	update_mob()
+	playsound(loc, 'sound/effects/empulse.ogg', 50, 1)
```
Comment (Sirryan2002) [inline code/modules/mod/modules/modules_security.dm:278]:
```suggestion
	playsound(loc, 'sound/effects/empulse.ogg', 50, TRUE)
```

File: code/modules/projectiles/guns/energy/special_eguns.dm [hunk 4]
```diff
@@ -879,3 +879,71 @@
...
+	. = ..()
+	if(ishuman(target))
+		return
+	if(isliving(target))
+		damage *= 4 //Up damage if not a human as we are not doing shenanigins
+		return
+	damage *= 6 //objects tend to fall apart as atoms are ripped up
+
+/obj/item/projectile/energy/vortex_blast/on_hit(atom/target, blocked = 0)
+	if((blocked < 100))
```
Comment (Sirryan2002) [inline code/modules/projectiles/guns/energy/special_eguns.dm:921]:
```suggestion
	if(blocked < 100)
```
also make this an early return

File: code/modules/projectiles/projectile_base.dm [hunk 5]
```diff
@@ -62,6 +62,8 @@
 	var/jitter = 0
 	/// Number of times an object can pass through an object. -1 is infinite
 	var/forcedodge = 0
+	/// Does the projectile increase fire stacks / immolate mobs on hit?
+	var/immolate = 0
```
Comment (Sirryan2002) [inline code/modules/projectiles/projectile_base.dm:66]: What does a higher value mean as opposed to 0 or 1?

File: code/modules/mob/living/simple_animal/hostile/mining/basilisk.dm (hunk 2 above)
Comment (Qwertytoforty) [reply code/modules/mob/living/simple_animal/hostile/mining/basilisk.dm:122]: Nope, will clarify that on the variable

File: code/modules/projectiles/projectile_base.dm (hunk 5 above)
Comment (Qwertytoforty) [reply code/modules/projectiles/projectile_base.dm:66]: this is the number of fire stacks the variable will apply, will make that more clear on  the autodoc

File: code/game/machinery/deployable.dm (hunk 1 above)
Comment (Qwertytoforty) [reply code/game/machinery/deployable.dm:331]: done

File: code/modules/projectiles/projectile_base.dm (hunk 5 above)
Comment (Qwertytoforty) [reply code/modules/projectiles/projectile_base.dm:66]: Done

**Review by Sirryan2002 — APPROVED** [review]

File: code/modules/mod/modules/modules_security.dm [hunk 6]
```diff
@@ -166,3 +166,126 @@
...
+	return grenade
+
+/obj/item/mod/module/anomaly_locked/firewall/prebuilt
+	prebuilt = TRUE
+	removable = FALSE // No switching it into another suit / no free anomaly core
+
+/// Vortex arm mounted shotgun. Fucks up reality in front of it, very power draining. Compeating with the vortex arm and stealth armor after all
+/obj/item/mod/module/anomaly_locked/vortex_shotgun
+	name = "MOD vortex shotgun module"
+	desc = "A module that uses a vortex core to rend the fabric of space time in front of it" //change
```
Comment (Henri215) [inline code/modules/mod/modules/modules_security.dm:203]:
```suggestion
	desc = "A module that uses a vortex core to rend the fabric of space time in front of it." //change
```
// change?

File: code/modules/projectiles/guns/energy/special_eguns.dm [hunk 7]
```diff
@@ -879,3 +879,72 @@
 
 #undef PLASMA_CHARGE_USE_PER_SECOND
 #undef PLASMA_DISCHARGE_LIMIT
+
+/obj/item/gun/energy/vortex_shotgun
+	name = "reality vortex wrist mounted shotgun"
+	desc = "some energy blast that shatters reality"
```
Comment (Henri215) [inline code/modules/projectiles/guns/energy/special_eguns.dm:885]:
```suggestion
	desc = "Some energy blast that shatters reality."
```

File: code/datums/uplink_items/uplink_nuclear.dm [hunk 8]
```diff
@@ -498,10 +498,13 @@
 	uplinktypes = list(UPLINK_TYPE_NUCLEAR, UPLINK_TYPE_SST)
 
 /datum/uplink_item/suits/hardsuit/shielded
```
Comment (Henri215) [inline code/datums/uplink_items/uplink_nuclear.dm:500]: This is no longer a hardsuit, update the path.

Comment (Qwertytoforty) [reply code/datums/uplink_items/uplink_nuclear.dm]: updated

**Review by S34NW — APPROVED** [review]

**Pull request merged**